File: IAIso-v5.0/core/iaiso-php/src/Policy/MaxAggregator.php
<?php

declare(strict_types=1);

namespace IAIso\Policy;

/** Maximum of per-execution pressures. */
final class MaxAggregator implements Aggregator
{
    public function name(): string { return 'max'; }

    public function aggregate(array $pressures): float
    {
        if (count($pressures) === 0) return 0.0;
        return (float) max($pressures);
    }
}

File: IAIso-v5.0/core/iaiso-java/iaiso-policy/src/main/java/io/iaiso/policy/MaxAggregator.java
package io.iaiso.policy;

import java.util.Map;

/** Maximum of per-execution pressures. */
public final class MaxAggregator implements Aggregator {
    @Override public String name() { return "max"; }
    @Override public double aggregate(Map<String, Double> p) {
        double max = 0.0;
        for (Double v : p.values()) {
            if (v > max) max = v;
        }
        return max;
    }
}
